docs/gadget-devel/gadget-ebpf-api.md

```diff
@@ -471,6 +471,60 @@ A new field containing the flags set will be added to the event:
 - `ebpf.formatter.enum`: Name of the new field. If the annotation is not set and the source field name has a `_raw` suffix, the target name will be set to the source name without that suffix.
 - `ebpf.formatter.bitfield.separator`: Separator used. Defaults to `|`.
 
+## Flexible Strings
+
+Some gadgets may need to use strings whose maximum size can be configured at
+runtime to save memory. For that, gadgets can use the `gadget_flex_string`
+type:
+
+```c
+#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
+#pragma clang attribute push(__attribute__((preserve_access_index)), \
+			     apply_to = record)
+#endif
```

Comment:
Please add a comment that can help understanding what these lines do. Does it affect other fields that are not `gadget_flex_string`? Can we help with this using a macro (e.g. `gadget_flex_string(mystring, MAX_SIZE)`)?